telemetry/telemetrybuffer.go

```diff
@@ -133,13 +128,24 @@ func (tb *TelemetryBuffer) StartServer() error {
 							}
 						} else {
 							telemetryLogger.Printf("Server closing client connection")
-							for index, value := range tb.connections {
+							var index int
+							var value net.Conn
+							var found bool
+
+							for index, value = range tb.connections {
 								if value == conn {
+									telemetryLogger.Printf("Server closing client connection2")
 									conn.Close()
-									tb.connections = remove(tb.connections, index)
-									return
```

Comment:
why did we change this?

Reply:
that is the right way..we should not remove array elements in iteration loop..also in one of the instances its crashed